### ISSUE TITLE
Bug - TreeView - Improve a11y experience

### ISSUE THREAD

**Issue by jessiehuff** [patternfly/patternfly#6478]
**Describe the problem**
This is an issue with 
- [x] Patternfly 5  
- [x] Patternfly 6  
- [ ] other

When Eric and I were reviewing the Tree View, we noticed some improvements that could be made ([summary of conversation here](https://github.com/patternfly/patternfly-org/pull/3916)).

- Mention `aria-multiselectable` is needed when the tree view is intended to have multiple items selected. So that'd be more a docs update in Core
- Add `aria-selected` to tree view items when selection made (we should probably look into whether `aria-checked` should be applied instead of `aria-selected` for checkbox tree views)
- Add aria-labeling to gives context to what the tree view is of/for (either `aria-labelledby` to a heading or `aria-label`)

**Expected behavior**
To match WAI ARIA tree view: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
